### DISCUSSION TITLE
Add Maintainers?

### DISCUSSION THREAD

**Discussion by karl-police** [Epix-Incorporated/Adonis#1194] [General]
😭

@Sceleratis

**Comment by moo1210** [comment]
do we really want it breaking more?

the whole updating system needs a revamp but I don't think that's ever going to happen tbh.

**Reply by karl-police** [reply]
I would be focusing on not breaking it, the entire opposite of breaking.

**Comment by Dimenpsyonal** [comment]
yeah!

**Comment by Dimenpsyonal** [comment]
> the whole updating system needs a revamp but I don't think that's ever going to happen tbh.

@moo1210 whst would you propose

**Comment by moo1210** [comment]
> > the whole updating system needs a revamp but I don't think that's ever going to happen tbh.
> 
> @moo1210 whst would you propose

Version numbers that make actually make sense (aka [semver](https://semver.org/)), updates that don't happen once or twice a year, yet when do happen, break half of Adonis. Unit testing that somewhat catches things breaking, possibly some type of feature flag system to toggle on/off changes without republishing the whole module (because we keep publishing "Hotfixes" after every update...) and rollouts for major changes, so we know if it actually affects someone or not. 

I used to for a while (and still somewhat want to) since Adonis 2.0 was announced wanted to make a pull request adding some of these things, namely the unit testing and a revamped Roact UI, but this project seems to just have somewhat, died in activity, stability, and quality in the past few years, to be honest. It would almost seem easier to just fork the repo and not deal with merging it back upstream at this point. 

I _somewhat_ liked the idea of Adonis 2.0 being a complete rewrite, which included most of these changes, but that also very quickly died off within a month or two. I'm also not a fan of having users manually migrate to 1.0 -> 2.0 for the end of time, I don't think we'll have viable adoption rates that way, either because of lack of knowing about Adonis 2.0, or just not wanting to go thru and edit config files and such. But, I'm not sure totally what's the best way to handle that either, since obviously, Roblox doesn't let you modify the config files in the studio without having a plugin installed (so perhaps a plugin _could_ be made that migrates it for you?), so they would still be using the old format even if they were on 2.0.

**Reply by karl-police** [reply]
Version Format 🤷

First people need to actually do the updates.

**Comment by Expertcoderz** [comment]
@Sceleratis has made it clear in an internal discussion that there shall be **no** new maintainers from the community. Except for select individuals with an extensive history of contributions who are handpicked on certain (rare) occasions for especially good reason.

**Comment by Dimenpsyonal** [comment]
> Version numbers that make actually make sense (aka [semver](https://semver.org/))

@moo1210 I mean, if you think this would be a good idea why not open an issue
